client/me/billing-history/transactions-header.jsx

```diff
@@ -149,85 +149,72 @@ class TransactionsHeader extends React.Component {
 		this.setState( { activePopover: activePopover } );
 	}
 
-	renderDatePicker( titleKey, titleTranslated, date, analyticsEvent ) {
-		const filter = { date };
-		const currentDate = this.props.filter.date || {};
-		let isSelected;
-
-		if ( date.newest ) {
-			isSelected = date.newest === currentDate.newest;
-		} else if ( date.month && currentDate.month ) {
-			isSelected = date.month.isSame( currentDate.month, 'month' );
-		} else if ( date.before ) {
-			isSelected = Boolean( currentDate.before );
-		} else {
-			isSelected = false;
-		}
-
+	renderDatePicker( titleKey, titleTranslated, value, count, analyticsEvent ) {
+		const currentDate = this.props.filter.date;
+		const isSelected = isEqual( currentDate, value );
 		analyticsEvent = 'undefined' === typeof analyticsEvent ? titleKey : analyticsEvent;
 
 		return (
 			<DropdownItem
 				key={ titleKey }
 				selected={ isSelected }
-				onClick={ this.getDatePopoverItemClickHandler( analyticsEvent, filter ) }
-				count={ date.newest ? null : this.getFilterCount( filter ) }
+				onClick={ this.getDatePopoverItemClickHandler( analyticsEvent, value ) }
+				count={ count }
 			>
 				{ titleTranslated }
 			</DropdownItem>
 		);
 	}
 
-	handlePickerSelection( filter ) {
-		this.setFilter( filter );
-		this.setState( { searchValue: '' } );
-	}
-
-	getFilterCount( filter ) {
-		if ( ! this.props.transactions ) {
-			return;
-		}
-
-		return tableRows.filter( this.props.transactions, filter ).length;
-	}
-
 	renderAppsPopover() {
+		const { filter, appFilters } = this.props;
+		const selectedFilter = find( appFilters, filterMeta => filterMeta.value === filter.app );
```

Comment:
We could shorten this to take advantage of `find`'s shorthand, so:

```
const selectedFilter = find( appFilters, filterMeta => filterMeta.value === filter.app );
```

would become something like:

```
const selectedFilter = find( appFilters, { value: filter.app } );
```